### PR TITLE
Fix OData methods with OData primitive return type

### PR DESCRIPTION
**Pull request by MIchaelMainer**
## Summary

ODate primitives method return types are returned in an object.
We weren't handling the object which resulted in serialization
exceptions.

	modified:   Templates/CSharp/Base/SharedCSharp.template.tt
	modified:   Templates/CSharp/Requests/MethodRequest.cs.tt

## Generated code differences

https://github.com/microsoftgraph/msgraph-beta-sdk-dotnet/pull/160
https://github.com/microsoftgraph/msgraph-sdk-dotnet/pull/800

## Command line arguments to run these changes

```pwsh
# Run against the current typewriter dev branch 
-v Info -m https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml -o dotnetbeta_old -l CSharp -g Files -e beta

# Run against this typewriter branch 
-v Info -m https://raw.githubusercontent.com/microsoftgraph/msgraph-metadata/master/clean_beta_metadata/cleanMetadataWithDescriptionsbeta.xml -o dotnetbeta_new -l CSharp -g Files -e beta
```

## Links to issues or work items this PR addresses

The generated changes are supported by these changes: https://github.com/microsoftgraph/msgraph-sdk-dotnet-core/pull/135

### CODE REVIEW

File: Templates/CSharp/Requests/MethodRequest.cs.tt [hunk 1]
```diff
@@ -176,10 +223,17 @@ namespace <#=@namespace#>
             return this.SendStreamRequestAsync(null, cancellationToken, httpCompletionOption);
 <#
     }
-    else if (!string.IsNullOrEmpty(sendAsyncReturnType))
+    else if (!string.IsNullOrEmpty(sendAsyncReturnType) && !isPrimitiveReturnType)
```
Comment (zengin) [inline Templates/CSharp/Requests/MethodRequest.cs.tt:226]: nit: [readability] consider nesting `isPrimitiveReturnType`
```csharp
else if (!string.IsNullOrEmpty(sendAsyncReturnType))
{
    if (isPrimitiveReturnType)
    {
        ...
    }
    else
    {
        ...
    }
}
```

File: Templates/CSharp/Requests/MethodRequest.cs.tt [hunk 2]
```diff
@@ -276,10 +330,17 @@ namespace <#=@namespace#>
...
```
Comment (zengin) [inline Templates/CSharp/Requests/MethodRequest.cs.tt:333]: same nesting comment.

File: test/Typewriter.Test/Given_a_valid_metadata_file_to_Typewriter.cs [hunk 3]
```diff
@@ -555,5 +555,145 @@ public void It_transforms_metadata()
             Assert.IsTrue(hasContainsTargetBeenSet, $"The expected ContainsTarget attribute wasn't set in the transformed cleaned metadata.");
             Assert.IsFalse(hasCapabilityAnnotations, $"The expected capability annotations weren't removed in the transformed cleaned metadata.");
         }
+
+        [Test, RunInApplicationDomain]
```
Comment (zengin) [inline test/Typewriter.Test/Given_a_valid_metadata_file_to_Typewriter.cs:559]: nit: tests can be combined with data attributes.

**Review by zengin — DISMISSED** [review]

File: Templates/CSharp/Requests/MethodRequest.cs.tt (hunk 1 above)
Comment (MIchaelMainer) [reply Templates/CSharp/Requests/MethodRequest.cs.tt]: Fixed with 922951b8

File: Templates/CSharp/Requests/MethodRequest.cs.tt (hunk 2 above)
Comment (MIchaelMainer) [reply Templates/CSharp/Requests/MethodRequest.cs.tt]: Fixed with 922951b8

File: test/Typewriter.Test/Given_a_valid_metadata_file_to_Typewriter.cs (hunk 3 above)
Comment (MIchaelMainer) [reply test/Typewriter.Test/Given_a_valid_metadata_file_to_Typewriter.cs:559]: Fixed with 405c1861

Comment (zengin) [reply test/Typewriter.Test/Given_a_valid_metadata_file_to_Typewriter.cs:559]: yay, less code! :) Thanks for taking the time to make these changes. Really appreciate it.

**Review by zengin — APPROVED** [review]

File: Templates/CSharp/Requests/IMethodRequest.cs.tt [hunk 4]
```diff
@@ -18,19 +18,42 @@ var requestType = entityName + methodName + "Request";
...
+if (returnEntityType != null) 
+{
+    returnEntityParameter = returnEntityType.ToLower();
+
+    // Updates to supported OData primitives need to occur here,
+    // IMethodRequest.cs.tt, Microsoft.Graph.Core, and in
+    // GetMethodRequestPrimitiveReturnTypeString() in SharedCSharp.
+    var tempReturnType = GetMethodRequestPrimitiveReturnTypeString(returnEntityType);
+
+    // These magic strings represent types in M.G.C. If the return
```
Comment (zengin) [inline Templates/CSharp/Requests/IMethodRequest.cs.tt:30]: nit: What it represents is clear from the context, but having it full string may help in a future search for this comment.

```suggestion
    // These magic strings represent types in Microsoft.Graph.Core. If the return
```

**Review by zengin — APPROVED** [review]

**Review by nikithauc — APPROVED** [review]

**Review by zengin — APPROVED** [review]

**Pull request merged**